src/ducks/subject.js

```diff
@@ -42,18 +44,68 @@ const subjectReducer = (state = initialState, action) => {
         currentSubject: action.currentSubject,
         status: SUBJECT_STATUS.READY,
         queue: action.queue,
+        favorite: action.favorite,
       });
 
     case FETCH_SUBJECT_ERROR:
       return Object.assign({}, state, {
         status: SUBJECT_STATUS.ERROR,
       });
 
+    case TOGGLE_FAVORITE:
+      return Object.assign({}, state, {
+        favorite: action.favorite,
+      });
+
     default:
       return state;
   }
 };
 
+const createFavorites = (project) => {
+  const links = {
+    subjects: [],
+    projects: [project.id],
+  };
+  const collection = {
+    favorite: true,
+    display_name: `Favorites ${project.data.display_name}`,
```

Comment:
Just one change requested: `${project.data.display_name}` should have a bit of existential checking, namely something along the lines of...
```
const display_name =  (project.data) ? project.data.display_name : 'UNKNOWN PROJECT';
```

At the moment, this PR works fine for our code because we have a functional project setup (see `const TEMPORARY_HARDCODED_PROJECT_ID = '1764';` in the code above) but - with the assumption that our code will eventually be ported for future projects - it'd be good if we add some robust safeties this early on.